### DISCUSSION TITLE
Selecting only active particles on density maps

### DISCUSSION THREAD

**Discussion by ocecept** [OpenDrift/opendrift#867] [Q&A]
Hi guys;

I have a simulation where I have set a one year continuous release with a max_age_seconds of 180 days. I run the model for 1.5 years. 

Now, I am trying to do a density map trying to show how many times a lat lon bin have been visited during my simulation period. 
I am able to do that following the "Analysing huge output files" gallery example. 

h = oa.get_histogram(pixelsize_m=10000)
b=h.isel(origin_marker=0).sum(dim='time')
oa.plot(background=b.where(b>0), fast=True, show_elements=False, vmin=0, vmax=1000, clabel='First seeding')

However, I you like to only use the "active" particles time in the calculations. 

I know, my status flags output are "active stranded retired" and I am trying to use it to find the density maps without success. 

Any suggestions where to start? 

Best regards;

Carlos

**Comment by knutfrode** [comment]
Hi,

status_flags `active stranded retired` correspond to 0, 1, 2 (may differ from case to case, except that active is always 0).
Thus you can try to also select where `status==0`

**Reply by ocecept** [reply]
HI Knutfrode; thanks for your replay. 

I tried to do selecting status==0 in the line b=h.isel(origin_marker=0).sum(dim='time') but 'status' is not exported to h in oa.get_histogram. 

Where do you suggest to  select where status==0? 

I tried to change get_histogram in basemodel to 

            h = histogram(self.ds.lon.where(self.ds.origin_marker == om & self.ds.status ==0),
                          self.ds.lat.where(self.ds.origin_marker == om & self.ds.status ==0),
                          bins=[lonbin, latbin],
                          dim=['trajectory'],
                          **kwargs)

but it did not work.

**Reply by ocecept** [reply]
I ended up finding the solution for my question in the "Show dominating source" example. I can select any status before doing the histogram.

    oa = opendrift.open_xarray(outfile)
    oa.ds = oa.ds.where(oa.ds.status == 0)
    h = oa.get_histogram(pixelsize_m=10000)
